### PR TITLE
Feat: Allow use of Account Number as Parent in Chart of Accounts Importer

### PR DESCRIPTION
**Pull request by erpjosephalba**
Chart of Accounts that use Account Numbers will be easier to import using Chart of Accounts Importer if Account Numbers are used as parent instead of Account Name.

Screenshot (11)

Files changed for this feature:
- erpnext/erpnext/accounts/doctype/account/chart_of_accounts/chart_of_accounts.py
- erpnext/erpnext/accounts/doctype/chart_of_accounts_importer/chart_of_accounts_importer.js
- erpnext/erpnext/accounts/doctype/chart_of_accounts_importer/chart_of_accounts_importer.json
- erpnext/erpnext/accounts/doctype/chart_of_accounts_importer/chart_of_accounts_importer.py

Screen Shots:

Screenshot (8)

Use Account Number as Parent check box is added. 
- If checked, Account Number is used as parent by Chart of Accounts Importer.
- otherwise, Account Name is used, like before.

Screenshot (16)

Import Successful

Note:
- I had to add an Account Ledger (child: Bank Account Number 1) for Bank Account group.

### CODE REVIEW

File: erpnext/accounts/doctype/account/chart_of_accounts/chart_of_accounts.py [hunk 1]
```diff
@@ -62,6 +62,60 @@ def _import_accounts(children, parent, root_type, root_account=False):
 		rebuild_tree("Account", "parent_account")
 		frappe.local.flags.ignore_on_update = False
 
+# allow use of account number as parent in Chart of Accounts importer		
+def create_charts_with_account_number(company, chart_template=None, existing_company=None, custom_chart=None):
```
Comment (nabinhait) [inline erpnext/accounts/doctype/account/chart_of_accounts/chart_of_accounts.py:66]: This almost duplicate code as above and should be commonified.

File: erpnext/accounts/doctype/account/chart_of_accounts/chart_of_accounts.py [hunk 2]
```diff
@@ -85,6 +139,17 @@ def identify_is_group(child):
 
 	return is_group
 
+# allow use of account number as parent in Chart of Accounts importer		
+def identify_is_group_with_account_number(child):
```
Comment (nabinhait) [inline erpnext/accounts/doctype/account/chart_of_accounts/chart_of_accounts.py:143]: This almost duplicate code as above and should be commonified.

File: erpnext/accounts/doctype/account/chart_of_accounts/chart_of_accounts.py [hunk 3]
```diff
@@ -232,3 +297,31 @@ def _import_accounts(children, parent):
 
 	_import_accounts(chart, None)
 	return accounts
+
+# allow use of account number as parent in Chart of Accounts importer		
+@frappe.whitelist()
+def build_tree_from_json_with_account_number(chart_template, chart_data=None):
```
Comment (nabinhait) [inline erpnext/accounts/doctype/account/chart_of_accounts/chart_of_accounts.py:303]: This almost duplicate code as above and should be commonified.

File: erpnext/accounts/doctype/chart_of_accounts_importer/chart_of_accounts_importer.py [hunk 4]
```diff
@@ -128,6 +137,73 @@ def return_parent(data, child):
 
 	return out
 
+def build_forest_with_account_number(data):
```
Comment (nabinhait) [inline erpnext/accounts/doctype/chart_of_accounts_importer/chart_of_accounts_importer.py:140]: Duplicate code, please commonify

**Comment by erpjosephalba** [comment]
I am closing this pull request to submit a different pull request which allows account number to be used as parent account in chart of accounts importer.

This new pull request will not have a checkbox Use Account Number as Parent in the doctype. Instead, the code will automatically detect if the imported csv file uses Account Name or Account Number as parent. It is already working. I will submit this later

**Pull request closed without merging**